thunder/executors/torch_autograd.py

```diff
@@ -60,6 +60,41 @@ def rename_bwd_trace_outputs(bwd_trace: TraceCtx, fwd_trace: TraceCtx) -> TraceC
     return renamed_bwd_trace
 
 
+# TODO: think about where to place it, seems useful
+class BasicTensorSubclass(torch.Tensor):
+    """
+    A tensor subclass that does own explicit storage (created with _make_wrapper_subclass),
+    has a reference to a torch.Tensor instance, and dispatches to torch.Tensor operations.
+    """
+
+    @staticmethod
+    def __new__(cls, t: torch.Tensor | None):
+        # Allow to propagate None - this is useful in the context of torch.autograd
+        if t is None:
+            return None
```

Comment:
saved_tensors is implemented to store only tensors and nothing else is allowed by torch.autograd.Function's ctx.save_for_backward method.

Reply:
Some failing tests begged to differ